include/odp/api/spec/packet.h

```diff
@@ -2140,6 +2140,50 @@ uint32_t odp_packet_payload_offset(odp_packet_t pkt);
  */
 int odp_packet_payload_offset_set(odp_packet_t pkt, uint32_t offset);
 
+/**
+ * Enable or disable Tx packet drop based on age
+ *
+ * Enable or disable Tx packet drop based on packet age. When enabled, packet will be dropped
+ * if it is in Tx pktout queue for longer than per-packet timeout.
+ *
+ * @param pkt     Packet handle
+ * @param enable  True indicates aging to be enabled on the packet.
+ */
+void odp_packet_aging_drop_set(odp_packet_t pkt, odp_bool_t enable);
+
+/**
+ * Set Tx packet aging timeout
+ *
+ * Set a packet aging timeout in usec which limits the age of packet in pktout queue or
+ * traffic scheduler/shapers. Packet exceeding its timeout will be dropped when packet aging
+ * drop is enabled.
```

Comment:
Maybe the exact point of packet drop should be specified (or alternatively, specified as implementation-specific). In particular, if a packet ages out when in TM queue, is the packet dropped from the queue right then and there, or does it still go through the schedulers and shapers in TM (affecting other traffic) and dropped at a later point?

Reply:
As discussed, I think we can leave this as implementation defined behavior as to when will it be dropped and what part of SoC bandwidth is consumed even if is about to be dropped.